File: Checklists_Code/Checklists_Code/DataModel.m
//
//  DataModel.m
//  Checklists_Code
//
//  Created by wangx on 15/6/26.
//  Copyright (c) 2015年 wxiang1991. All rights reserved.
//

#import "DataModel.h"

@implementation DataModel
-(instancetype)init{
    self=[super init];
    if (self) {
        [self loadChecklists];
    }
    return self;
}

-(void)sortChecklists{
    [self.lists sortUsingSelector:@selector(compare:)];
}

#pragma mark 获取沙盒地址
-(NSString*)documentsDirectory{
    NSArray *paths = NSSearchPathForDirectoriesInDomains(NSDocumentDirectory, NSUserDomainMask, YES);
    NSString *documentsDirectory = [paths firstObject];
    return documentsDirectory;
}

-(NSString*)dataFilePath{
    return [[self documentsDirectory]stringByAppendingPathComponent:@"Checklists.plist"];
}

-(void)saveChecklists{
    NSMutableData *data = [[NSMutableData alloc]init];
    NSKeyedArchiver *archiver = [[NSKeyedArchiver alloc]initForWritingWithMutableData:data];
    [archiver encodeObject:_lists forKey:@"Checklists"];
    [archiver finishEncoding];
    [data writeToFile:[self dataFilePath] atomically:YES];
}

-(void)loadChecklists{
    NSString *path = [self dataFilePath];
    if([[NSFileManager defaultManager]fileExistsAtPath:path]){
        NSData *data = [[NSData alloc]initWithContentsOfFile:path];
        NSKeyedUnarchiver *unarchiver = [[NSKeyedUnarchiver alloc]initForReadingWithData:data];
        self.lists = [unarchiver decodeObjectForKey:@"Checklists"];
        [unarchiver finishDecoding];
    }else{
        self.lists = [[NSMutableArray alloc]initWithCapacity:20];
    }
    
}
@end

File: ChecklistSwift/ChecklistSwift/DataModel.swift
//
//  DataModel.swift
//  ChecklistSwift
//
//  Created by 王享 on 15/12/25.
//  Copyright © 2015年 王享. All rights reserved.
//

import UIKit

class DataModel: NSObject {
    
    var lists:Array<Checklist>?
    
    override init(){
        super.init()
        self.loadChecklists()
    }
    
    func documentsDirectory() -> String{
        if let dirs:[String] = NSSearchPathForDirectoriesInDomains(FileManager.SearchPathDirectory.documentDirectory, FileManager.SearchPathDomainMask.allDomainsMask, true) {
            let dir:String = dirs[0]
            return dir;
        }
    }
    func dataFilePath() -> String{
        let path = self.documentsDirectory()
        return path + "/Checklists.plist";
    }
    func saveChecklists(){
        let data:NSMutableData = NSMutableData()
        let archiver:NSKeyedArchiver = NSKeyedArchiver.init(forWritingWith: data)
        archiver.finishEncoding()
        data.write(toFile: self.dataFilePath(), atomically:true)
    }
    
    func loadChecklists() {
        let path:String = self.dataFilePath()
        if FileManager.default.fileExists(atPath: path){
            let data:Data = try! Data(contentsOf: URL(fileURLWithPath: path));
            let unarchiver:NSKeyedUnarchiver = NSKeyedUnarchiver(forReadingWith: data)
            self.lists = unarchiver.decodeObject(forKey: "Checklists") as? Array<Checklist>
            unarchiver.finishDecoding()
        }else{
            let checklist:Checklist = Checklist();
            self.lists = Array<Checklist>(repeating: checklist,count: 1);
        }
    }
}
